src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs

```diff
@@ -433,6 +436,34 @@ private void exportTextBlock_MouseLeftButtonDown(object sender, MouseButtonEvent
                        ex.Message, Res.ImportSettingsFailedMessage, MessageBoxButton.OK, MessageBoxImage.Exclamation);
                 }
             }
-        }        
+        }
+
+        // Show File path dialog
+        private void OnRequestShowFileDialog(object sender, EventArgs e)
+        {
+            var args = e as PythonTemplatePathEventArgs;
+            args.Cancel = true;
+
+            var dialog = new System.Windows.Forms.OpenFileDialog
+            {
+                // Navigate to initial folder.
+                FileName = args.Path
+            };
+
+            dialog.Filter = "All Files (*.*)|*.*";
```

Comment:
Do you think we should be limiting this to only files with .py extension?

Reply:
@reddyashish agreed - if this is specifically for Python we should explicitly limit it to that 😄